lua/nvim-tree/notify.lua

```diff
@@ -0,0 +1,44 @@
+local notify = {}
+
+local config = {
+  default_level = vim.log.levels.INFO,
+}
+
+local modes = {
+  { name = "trace", level = vim.log.levels.TRACE },
+  { name = "debug", level = vim.log.levels.DEBUG },
+  { name = "info", level = vim.log.levels.INFO },
+  { name = "warn", level = vim.log.levels.WARN },
+  { name = "error", level = vim.log.levels.ERROR },
+}
+
+do
```

Comment:
yeah, it needs to be done this way in case the module gets called before the setup function.

Reply:
As I mentioned in https://github.com/nvim-tree/nvim-tree.lua/pull/1693#discussion_r1008658840, I prefer not dealing with that API here since it's a bit more nuanced than what's suggested in https://github.com/nvim-tree/nvim-tree.lua/files/9894883/0001-feat-configurable-notification-level-consistent-setu.patch.txt

Problems with the proposed changes:
- it's not reusing the `config.notify` table (which was created for this specific purpose)
- it's hard-coding the notify handler regardless
- users can already override `vim.notify` so there's no point in creating yet another redirection layer on top of it
- and it's (highly) unlikely that a user would have two separate wrappers for `vim.notify` at the same time

my idea of an alternative source was something similar to how `|vim.lsp.util.get_progress_messages()|` works, where a status message can be displayed anywhere.

I'm happy to create a separate PR to track that :)

Reply:
That's fair enough. This is an unusual module.

We will be revisiting this shortly anyway.